src/Algorithms/threshold/Dockerfile

```diff
@@ -0,0 +1,19 @@
+# Define an image to start from 
+#FROM bethcimini/docker-gradio-demo:latest
+FROM gradio-base:latest
+
+# Set the working directory within the container
+WORKDIR /workspace
+
+# Install the dependencies for the gradio app
+RUN python -m pip install scipy==1.9.1 scikit-image==0.20.0
+
+# Add app.py from the directory where `docker build` was run and add it into the
+# workspace directory within the container
+ADD /Algorithms /workspace/
+
+# Expose port 8000. You CAN do this from the command line, but it makes it possible to run entirely in the app
+EXPOSE 8000
+
+# Run the app. Will be accessible at http://localhost:8000/
+CMD [ "python3" , "/workspace/Algorithms/threshold.py" ]
```

Comment:
This should be `app.py` not `threshold.py`, as the thing that gets run. But also much more importanly the `threshold` algorithm is the thing that gets wrapped, not the thing that is doing the wrapping.

We flipped on this at the tail end of discussion #5 .

It should be gradio docker wraps threshold docker.

Reply:
I understood that `gradio(Interface)` is going to wrap `threshold(algorithm)`, then the first thing which is going to run from `CI/CD` should be `threshold`, I mean the `Algorithm` docker. So, isn't that this docker file shouldn't run `app.py`. As it will be generated later, by the Interface Docker, which is inside the `Build` folder?

Reply:
I believe, this issue is resolved as we discussed that we don't want to run anything thru this docker-image. It's just for installing dependencies specific to algorithm, eg. scipy. Hence, we shouldn't `EXPOSE` the port here and don't need to have `CMD`

Reply:
I do think we want to generally set CMD as good practice so that the container is otherwise-reusable on its own, but agreed it's not needed for this